releasenotes-builder/src/main/java/com/github/checkstyle/publishers/SourceforgeRssPublisher.java

```diff
@@ -0,0 +1,113 @@
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.github.checkstyle.publishers;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Class for RSS post publication.
+ * @author Vladislav Lisetskii
+ */
+public class SourceforgeRssPublisher {
+
+    /** URL to send POST request to. */
+    private static final String POST_URL =
+            "https://sourceforge.net/rest/p/checkstyle/news";
+    /** Template for the post. */
+    private static final String POST_TEMPLATE =
+            "access_token=%s&title=Checkstyle %s&text=%s&state=published";
+
+    /** Patter for finding a number. */
+    private static final Pattern NUMBER_PATTERN = Pattern.compile("\\d+");
+
+    /** The name of the file to get post text from. */
+    private final String postFilename;
+
+    /** Token for posting. */
+    private final String bearerToken;
+    /** Release number. */
+    private final String releaseNumber;
+
+    /**
+     * Default constructor.
+     * @param postFilename the name of the file to get post text from.
+     * @param bearerToken token for posting.
+     * @param releaseNumber release number.
+     */
+    public SourceforgeRssPublisher(String postFilename, String bearerToken, String releaseNumber) {
+        this.postFilename = postFilename;
+        this.bearerToken = bearerToken;
+        this.releaseNumber = releaseNumber;
+    }
+
+    /**
+     * Publish release notes.
+     * @throws IOException if problem with access to files appears.
+     */
+    public void publish() throws IOException {
+        final int postsCountBefore = getPostsCount();
+
+        final HttpURLConnection conn = (HttpURLConnection) new URL(POST_URL).openConnection();
+        conn.setDoOutput(true);
+        conn.setRequestMethod("POST");
+        conn.setRequestProperty("Content-Type", "application/x-www-form-urlencoded");
+
+        final OutputStream os = conn.getOutputStream();
+        final String postText = new String(Files.readAllBytes(Paths.get(postFilename)),
+                StandardCharsets.UTF_8);
+        os.write(String.format(POST_TEMPLATE, bearerToken, releaseNumber, postText)
+                .getBytes(StandardCharsets.UTF_8));
+        os.flush();
+
+        final int responseCode = conn.getResponseCode();
+        conn.disconnect();
+        if (responseCode != HttpURLConnection.HTTP_OK && getPostsCount() == postsCountBefore) {
```

Comment:
please make make a single line comment on why you do testing for post counts, it is far from obvious.